appinventor/appengine/src/com/google/appinventor/client/wizards/youngandroid/NewYoungAndroidProjectWizard.java

```diff
@@ -55,11 +55,13 @@ public NewYoungAndroidProjectWizard(final ProjectToolbar toolbar) {
     projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel(), new Validator() {
       @Override
       public boolean validate(String value) {
+        value = value.trim();
```

Comment:
So one problem I noticed with calling trim here is that we don't see a warning for a string that contains a single sequence of characters with whitespace on either side. Only if you have at least two words (e.g., "Hello World") does the check do the expected behavior. Consider moving the trim() call inside of the getWarningMessage check as I think that would solve it. To test, type a bunch of spaces, a single word, then a bunch more spaces. This should yield a warning, but it doesn't currently.